runtime/ms_rest_azure/spec/polling_state_spec.rb

```diff
@@ -10,25 +10,31 @@ module MsRestAzure
   describe PollingState do
     it 'should initialize status from flattened response body' do
       response_body = double('response_body', :provisioning_state => 'InProgress')
-      response = double('response',
```

Comment:
could there be some type of inconsistency between the response content and provisioning state, and if there is, how does our code handle it? we should be returning some kind of error if something does not match up, but try to avoid failing with a null exception or similar, so just something to check on.